.cbmc-batch/stubs/evp_md_ctx_is_valid_stub.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <ec_utils.h>
+#include <make_common_data_structures.h>
+#include <openssl/evp.h>
+#include <openssl/hmac.h>
+#include <openssl/kdf.h>
+#include <openssl/rsa.h>
+#include <proof_helpers/nondet.h>
+
+/* Abstraction of the EVP_MD_CTX struct */
+struct evp_md_ctx_st {
+    bool is_initialized;
+    EVP_PKEY *pkey;
+    size_t digest_size;
+};
+
+/* Helper function for CBMC proofs: checks if EVP_MD_CTX is valid.
+ * Use this stub when we are *certain* there is no pkey associated with the digest context.
+ */
+bool evp_md_ctx_is_valid(EVP_MD_CTX *ctx) {
+    return ctx && ctx->is_initialized && ctx->digest_size <= EVP_MAX_MD_SIZE && ctx->pkey == NULL;
```

Comment:
why is null the only valid pkey

Reply:
In general, it is not and the more generic evp_md_ctx_is_valid function reflects this.  Similarly as in https://github.com/aws/aws-encryption-sdk-c/pull/582#discussion_r478662276, I introduce this stub to improve coverage in the case where we are certain that pkey == NULL for the function being addressed in the CBMC proof. This greatly improves coverage. Once again, I have renamed these stubs to make this assumption more clear.